drivers/platform/xilinx/irq.c

```diff
@@ -238,15 +256,27 @@ int32_t irq_source_disable(struct irq_ctrl_desc *desc, uint32_t irq_id)
  * @return SUCCESS in case of success, FAILURE otherwise.
  */
 int32_t irq_register(struct irq_ctrl_desc *desc, uint32_t irq_id,
-		     void (*irq_handler)(void *data), void *dev_instance)
+		     void (*irq_handler)(void *context, void *extra),
```

Comment:
If you need multiple parameters (eg context and extra for the irq handler) you can pack them in a structure and pass it as reference (*data).